Firebase/Core/FIRBundleUtil.m

```diff
@@ -52,10 +52,10 @@ + (BOOL)hasBundleIdentifierPrefix:(NSString *)bundleIdentifier inBundles:(NSArra
     // This allows app extensions that have the app's bundle as their prefix to pass this test.
     NSString *applicationBundleIdentifier =
         [GULAppEnvironmentUtil isAppExtension]
-            ? [self bundleIdentifierByRemovingLastPartFrom:bundleIdentifier]
-            : bundleIdentifier;
+            ? [self bundleIdentifierByRemovingLastPartFrom:bundle.bundleIdentifier]
+            : bundle.bundleIdentifier;
 
-    if ([applicationBundleIdentifier isEqualToString:bundle.bundleIdentifier]) {
+    if ([applicationBundleIdentifier isEqualToString:bundleIdentifier]) {
```

Comment:
Shouldn't we here check for a prefix as the method name suggests?
This seems like it would only pass if appBundleID and extensionBundleID are the same which would never be the case.

```
-    if ([applicationBundleIdentifier isEqualToString:bundleIdentifier]) {
+    if ([bundleIdentifier hasPrefix:applicationBundleIdentifier]) {
```